api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -99,6 +99,20 @@ public interface RewriteDataFiles
 
   boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = true;
 
+  /**
+   * Removes dangling delete files from current snapshot after compaction. A delete file is
+   * considered dangling if it does not apply to any non-expired data file.
```

Comment:
Rewrite operation never used to delete any files. So, deleting some files from rewrite might confuse users? 

Is it possible to clean these files from expire snapshots (or as orphan files). 
Maybe we can just update the metadata by making these unreferenced and the actual delete can happen from GC functions.

Reply:
"delete files" here means delete files from current(live) snapshot. Rewrite data files can already delete equality delete files from current snapshot. It doesn't remove files from filesystem.

Reply:
"delete files" here means delete files from current(live) snapshot. Rewrite data files can already delete equality delete files from current snapshot. It doesn't remove files from filesystem.

Reply:
>  A delete file is considered dangling if it does not apply to any non-expired data file.
I think the most accurate wording is `if it does not apply to data files of current snapshot`.